trie/trie.go

```diff
@@ -101,7 +101,7 @@ func New(root common.Hash, db *Database) (*Trie, error) {
 		db:           db,
 		originalRoot: root,
 	}
-	if (root != common.Hash{}) && root != emptyRoot {
+	if (root != common.Hash{}) && (root != emptyRoot) {
```

Comment:
This one's a bit weird (the original was too). Please change to `if root != (common.Hash{}) && root != emptyRoot {`

Reply:
Thanks for your reply, fixed ans force pushed.